src/main/js/component/AboutInfo.js

```diff
@@ -39,36 +43,89 @@ class AboutInfo extends React.Component {
                     headerContainerClass="scrollable"
                     bodyContainerClass="scrollable"
                 >
-                    <TableHeaderColumn dataField="label" isKey dataFormat={nameRenderer}>
+                    <TableHeaderColumn dataField="name" isKey dataFormat={nameRenderer}>
                         Name
                     </TableHeaderColumn>
                     <TableHeaderColumn dataField="urlName" hidden>
                         Url
                     </TableHeaderColumn>
+                    <TableHeaderColumn dataField="navigate" hidden>
+                        Navigate
+                    </TableHeaderColumn>
                 </BootstrapTable>
             </div>
         );
     }
 
+    createProviderTableData(globalDescriptors) {
+        const data = [];
+        for (let key in globalDescriptors) {
+            const descriptor = globalDescriptors[key];
+            if (!data.find(existing => existing.urlName === descriptor.urlName)) {
+                data.push({
+                    name: descriptor.label,
+                    urlName: descriptor.urlName,
+                    navigate: true
+                });
+            }
+        }
+        return data;
+    }
+
+    createChannelTableData(globalDescriptors, distributionDescriptors) {
+        const data = [];
+        const nameSet = new Set();
+        globalDescriptors.map(descriptor => descriptor.name)
+        .forEach(name => nameSet.add(name));
+        distributionDescriptors.map(descriptor => descriptor.name)
+        .forEach(name => nameSet.add(name));
+
+        nameSet.forEach(descriptorName => {
+            const globalDescriptor = globalDescriptors.find(descriptor => descriptor.name === descriptorName);
+            const distributionDescriptor = distributionDescriptors.find(descriptor => descriptor.name === descriptorName);
```

Comment:
Should this be done in the else statement? It doesn't look like it's used anywhere else.

Reply:
It would save a little processing time.